projects/igniteui-angular/src/lib/directives/text-selection/text-selection.directive.spec.ts

```diff
@@ -12,57 +12,122 @@ describe('IgxSelection', () => {
-            expect(input.value.substring(input.selectionStart, input.selectionEnd)).toEqual(input.value);
-        });
+        expect(input.selectionEnd).toEqual(input.value.length);
+        expect(input.value.substring(input.selectionStart, input.selectionEnd)).toEqual(input.value);
     });
 
-    it('Should select the text when the input is clicked', async () => {
+    it('Should select the text when the input is clicked', ()=> {
         const fix = TestBed.createComponent(TriggerTextSelectionOnClickComponent);
         fix.detectChanges();
 
         const input: DebugElement = fix.debugElement.query(By.css('input'));
         const inputNativeElem = input.nativeElement;
         const inputElem: HTMLElement = input.nativeElement;
 
-        inputElem.click();
+        inputElem.click(); // might need to change to .focus
+        fix.detectChanges();
+
+        expect(inputNativeElem.selectionEnd).toEqual(inputNativeElem.value.length);
+        expect(inputNativeElem.value.substring(inputNativeElem.selectionStart, inputNativeElem.selectionEnd))
+            .toEqual(inputNativeElem.value);
+    });
+
+    it('Should check if the value is selected if based on input type', () => {
+        const fix = TestBed.createComponent(TriggerTextSelectionOnClickComponent);
+        const selectableTypes: Types[] = [
+            { "text" : "Some Values!" },
+            { "search": "Search!" },
+            { "password": "********" },
+            { "tel": '+(359)554-587-415' },
+            { "url": "www.infragistics.com" },
+            { "number": 2136512312 }
+        ];
+
+        const nonSelectableTypes: Types[] = [
+            {'date': new Date() },
+            {'datetime-local': "2018-06-12T19:30" },
+            {'email': 'JohnSmith@gmail.com'},
+            {'month': "2018-05" },
+            {'time': "13:30"},
+            {'week': "2017-W01"}
+        ];
+
+        //skipped on purpose, if needed feel free to add to any of the above categories
+        //const irrelevantTypes = ['button', 'checkbox', 'color', 'file', 'hidden', 'image', 'radio', 'range', 'reset', 'submit']
+
+        const input = fix.debugElement.query(By.css('input'));
+        const inputNativeElem = input.nativeElement;
+        const inputElem: HTMLElement = input.nativeElement;
+
+        selectableTypes.forEach( el => {
+            let type = Object.keys(el)[0];
+            let val = el[type];
+            fix.componentInstance.inputType = type;
+            fix.componentInstance.inputValue = val;
+            fix.detectChanges();
 
-        fix.whenStable().then(() => {
+            inputElem.click();
             fix.detectChanges();
-            expect(inputNativeElem.selectionEnd).toEqual(inputNativeElem.value.length);
-            expect(inputNativeElem.value.substring(inputNativeElem.selectionStart, inputNativeElem.selectionEnd))
-                .toEqual(inputNativeElem.value);
+
+            if(type !== 'number' && type !== 'tel'){
```

Comment:
Test the `tel` type. We are testing everything but `number` and `tel` type here. Then we are testing `number` but we never test `tel`.